extensions/BeatBlox/index.js

```diff
@@ -243,11 +285,32 @@
             return base + delta;
         }
 
+        function durationToBeats(duration, durationSpecial = '') {
+            let playDuration = availableNoteDurations[duration];
+            
+            if (!durationSpecial) {
+                // For regular durations
+                return 4 / playDuration;
+            } else {
+                // For special durations
+                playDuration = availableNoteDurations[durationSpecial + duration];
+                
+                if (playDuration === 2/3 || playDuration === 4/7) {
```

Comment:
These give different results than before (and the 2/3 and 4/7 ones give negative durations).